File: src/main/java/no/nav/syfo/services/TilgangskontrollService.java
package no.nav.syfo.services;

import no.nav.syfo.tilgang.BrukerTilgangConsumer;
import org.springframework.stereotype.Service;

import javax.inject.Inject;

@Service
public class TilgangskontrollService {

    private final BrukerTilgangConsumer brukerTilgangConsumer;

    @Inject
    public TilgangskontrollService(
            BrukerTilgangConsumer brukerTilgangConsumer
    ) {
        this.brukerTilgangConsumer = brukerTilgangConsumer;
    }

    public boolean sporOmNoenAndreEnnSegSelvEllerEgneAnsatte(String innloggetFnr, String oppslattFnr) {
        return !(sporInnloggetBrukerOmSegSelv(innloggetFnr, oppslattFnr) || sporInnloggetBrukerOmEnAnsatt(oppslattFnr));
    }

    private boolean sporInnloggetBrukerOmEnAnsatt(String oppslattFnr) {
        return brukerTilgangConsumer.hasAccessToAnsatt(oppslattFnr);
    }

    private boolean sporInnloggetBrukerOmSegSelv(String innloggetFnr, String fnr) {
        return fnr.equals(innloggetFnr);
    }
}

File: src/main/kotlin/no/nav/syfo/tilgang/TilgangskontrollService.kt
package no.nav.syfo.tilgang

import no.nav.syfo.tilgang.consumer.BrukerTilgangConsumer
import org.springframework.stereotype.Service
import javax.inject.Inject

@Service
class TilgangskontrollService @Inject constructor(
        private val brukerTilgangConsumer: BrukerTilgangConsumer
) {
    fun sporOmNoenAndreEnnSegSelvEllerEgneAnsatte(innloggetFnr: String, oppslattFnr: String): Boolean {
        return !(sporInnloggetBrukerOmSegSelv(innloggetFnr, oppslattFnr) || sporInnloggetBrukerOmEnAnsatt(oppslattFnr))
    }

    private fun sporInnloggetBrukerOmEnAnsatt(oppslattFnr: String): Boolean {
        return brukerTilgangConsumer.hasAccessToAnsatt(oppslattFnr)
    }

    private fun sporInnloggetBrukerOmSegSelv(innloggetFnr: String, fnr: String): Boolean {
        return fnr == innloggetFnr
    }

}